src/edge_utils.c

```diff
@@ -2288,92 +2288,137 @@ static int routectl(int cmd, int flags, n2n_route_t *route, int if_idx) {
+            eee->sn_route_to_clean = calloc(1, sizeof(n2n_route_t));
 
-      /* ip route add 0.0.0.0/1 via n2n_gateway */
-      custom_route.net_addr = 0;
-      custom_route.net_bitlen = 1;
-      custom_route.gateway = route->gateway;
+            /* Store a copy of the rules into the runtime to delete it during shutdown */
+            if (eee->sn_route_to_clean)
+                *eee->sn_route_to_clean = custom_route;
 
-      if(routectl(RTM_NEWROUTE, NLM_F_CREATE | NLM_F_EXCL, &custom_route, eee->device.if_idx) < 0)
-	return(-1);
+            /* ip route add 0.0.0.0/1 via n2n_gateway */
+            custom_route.net_addr = 0;
+            custom_route.net_bitlen = 1;
+            custom_route.gateway = route->gateway;
 
-      /* ip route add 128.0.0.0/1 via n2n_gateway */
-      custom_route.net_addr = 128;
-      custom_route.net_bitlen = 1;
-      custom_route.gateway = route->gateway;
+            if (routectl(RTM_NEWROUTE, NLM_F_CREATE | NLM_F_EXCL, &custom_route, eee->device.if_idx) < 0)
+                return(-1);
 
-      if(routectl(RTM_NEWROUTE, NLM_F_CREATE | NLM_F_EXCL, &custom_route, eee->device.if_idx) < 0)
-	return(-1);
-    } else {
-      /* ip route add net via n2n_gateway */
-      if(routectl(RTM_NEWROUTE, NLM_F_CREATE | NLM_F_EXCL, route, eee->device.if_idx) < 0)
-	return(-1);
+            /* ip route add 128.0.0.0/1 via n2n_gateway */
+            custom_route.net_addr = 128;
+            custom_route.net_bitlen = 1;
+            custom_route.gateway = route->gateway;
+
+            if (routectl(RTM_NEWROUTE, NLM_F_CREATE | NLM_F_EXCL, &custom_route, eee->device.if_idx) < 0)
+                return(-1);
+        }
+        else {
+            /* ip route add net via n2n_gateway */
+            if (routectl(RTM_NEWROUTE, NLM_F_CREATE | NLM_F_EXCL, route, eee->device.if_idx) < 0)
+                return(-1);
+        }
     }
-  }
 #endif
 
-  return(0);
+    return(0);
+}
+
+static int edge_init_routes_win(n2n_edge_t *eee, n2n_route_t *routes, uint16_t num_routes)
+{
+#ifdef WIN32
+    int i;
+    struct in_addr net_addr, gateway;
+    char c_net_addr[32];
+    char c_gateway[32];
+    char cmd[256];
+
+    for (i = 0; i < num_routes; i++)
+    {
+        n2n_route_t *route = &routes[i];
+        if ((route->net_addr == 0) && (route->net_bitlen == 0))
+        {
+            traceEvent(TRACE_NORMAL, "Warning: The 0.0.0.0/0 route settings are not supported on Windows");
+            return (-1);
+        }
+        else
+        {
+            /* ip route add net via n2n_gateway */
+            memcpy(&net_addr, &(route->net_addr), sizeof(net_addr));
+            memcpy(&gateway, &(route->gateway), sizeof(gateway));
+            _snprintf(c_net_addr, sizeof(c_net_addr), inet_ntoa(net_addr));
+            _snprintf(c_gateway, sizeof(c_gateway), inet_ntoa(gateway));
+            _snprintf(cmd, sizeof(cmd), "route add %s/%d %s > nul", c_net_addr, route->net_bitlen, c_gateway);
```

Comment:
Why so many temporary variables, instead of simplifying the statement?

Or it can be like this:

_snprintf(cmd, sizeof(cmd), "route add %s/%d %s > nul", inet_ntoa(*((struct in_addr*)(&route->net_addr))), route->net_bitlen, inet_ntoa(*((struct in_addr*)(&route->gateway))));